pkg/limayaml/validate.go

```diff
@@ -372,6 +372,45 @@ func Validate(y *LimaYAML, warn bool) error {
 		}
 	}
 
+	if y.Rosetta.Enabled != nil {
+		if *y.Rosetta.Enabled && *y.VMType != VZ {
+			return fmt.Errorf("field `rosetta.enabled` can only be enabled for VMType %q; got %q", VZ, *y.VMType)
+		}
+	}
+
+	if y.NestedVirtualization != nil && *y.NestedVirtualization {
+		if *y.VMType != VZ {
+			return fmt.Errorf("field `nestedVirtualization` can only be enabled for VMType %q; got %q", VZ, *y.VMType)
+		}
+	}
+
+	if y.MountType != nil {
+		if err := validateMountType(y); err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+func validateMountType(y *LimaYAML) error {
+	validMountTypes := map[string]bool{
+		REVSSHFS: true,
+		NINEP:    true,
+		VIRTIOFS: true,
+		WSLMount: true,
+	}
+
+	if !validMountTypes[*y.MountType] {
+		return fmt.Errorf("field `mountType` %s, valid options are: %s, %s, %s, %s", *y.MountType, REVSSHFS, NINEP, VIRTIOFS, WSLMount)
+	}
+
+	if *y.MountType == VIRTIOFS && runtime.GOOS == "darwin" && IsNativeArch(AARCH64) {
+		if y.VMType == nil || *y.VMType != VZ {
+			return fmt.Errorf("field `mountType` requires vmType 'vz' on macOS (darwin); got %s", *y.VMType)
+		}
+	}
```

Comment:
Why are you checking for `AARCH64`? The requirement for `VZ` also exists for Intel.

And `y.VMType` cannot be `nil` here.

```suggestion
	if *y.MountType == VIRTIOFS && runtime.GOOS == "darwin" && *y.VMType != VZ {
			return fmt.Errorf("field `mountType` %q on macOS requires vmType %q; got %q", *y.MountType, VZ, *y.VMType)
		}
	}
```